store/mongo/datastore_mongo.go

```diff
@@ -535,6 +535,71 @@ func (db *DataStoreMongo) UpdateDevicesGroup(
 	}, nil
 }
 
+func (db *DataStoreMongo) GetFiltersAttributes(ctx context.Context) ([]model.FilterAttribute, error) {
+	database := db.client.Database(mstore.DbFromContext(ctx, DbName))
+	collDevs := database.Collection(DbDevicesColl)
+
+	const DbCount = "count"
+
+	cur, err := collDevs.Aggregate(ctx, []bson.M{
+		{
+			"$project": bson.M{
+				"attributes": bson.M{
+					"$objectToArray": "$" + DbDevAttributes,
+				},
+			},
+		},
+		{
+			"$unwind": "$" + DbDevAttributes,
+		},
+		{
+			"$project": bson.M{
+				DbDevAttributesName:  "$" + DbDevAttributes + ".v." + DbDevAttributesName,
+				DbDevAttributesScope: "$" + DbDevAttributes + ".v." + DbDevAttributesScope,
+			},
+		},
+		{
+			"$group": bson.M{
+				DbDevId: bson.M{
+					DbDevAttributesName:  "$" + DbDevAttributesName,
+					DbDevAttributesScope: "$" + DbDevAttributesScope,
+				},
+				DbCount: bson.M{
+					"$sum": 1,
+				},
+			},
+		},
+		{
+			"$sort": bson.D{
```

Comment:
I was going to comment on the `bson.M` here before you fixed it, but generally I think it's a good habit to discourage use of `bson.M` everywhere as it can have unintended random side-effects elsewhere as well.

Reply:
Well noted